lib/features/children/add_member/pages/failed_vpc_bottomsheet.dart

```diff
@@ -0,0 +1,145 @@
+                        ],
+                      ),
+                content: (state.status == CachedMembersStateStatus.loading)
+                    ? const SizedBox.shrink()
+                    : BodyMediumText(
+                        'We couldn’t take the \$0.50 for verification and the \$${walletAmount.toStringAsFixed(0)} for your child’s wallet.\n\nCheck your payment details and try again or choose another one.',
+                        textAlign: TextAlign.center,
+                      ),
+                primaryButton: GivtElevatedButton(
+                  isDisabled: state.status ==
+                          CachedMembersStateStatus.loading ||
+                      state.status == CachedMembersStateStatus.noFundsSuccess,
+                  text: 'Try again',
+                  amplitudeEvent:
+                      AmplitudeEvents.changePaymentMethodForFailedVPCClicked,
+                  onTap: () async {
+                    await cacheCubit.tryCreateMembersFromCache(members);
+                  },
+                ),
+                secondaryButton: GivtElevatedSecondaryButton(
+                  isDisabled: state.status ==
+                          CachedMembersStateStatus.loading ||
+                      state.status == CachedMembersStateStatus.noFundsSuccess,
+                  text: 'Change payment method',
+                  rightIcon: const FaIcon(
+                    FontAwesomeIcons.arrowsRotate,
+                    size: 24,
+                  ),
+                  onTap: () async {
+                    if (!context.mounted) return;
+                    await context.read<StripeCubit>().fetchSetupIntent();
+
+                    if (!context.mounted) return;
+
+                    try {
+                      await StripeHelper(context).showPaymentSheet();
+
+                      if (!context.mounted) return;
+                      await context.read<AuthCubit>().refreshUser();
+                    } on StripeException catch (e, stackTrace) {
+                      await AnalyticsHelper.logEvent(
+                        eventName: AmplitudeEvents.tryAgainForFailedVPCClicked,
+                      );
+
+                      /* Logged as info as stripe is giving exception
+                                         when for example people close the bottomsheet.
+                                         So it's not a real error :)
+                                      */
+                      LoggingInfo.instance.info(
+                        e.toString(),
+                        methodName: stackTrace.toString(),
+                      );
+                    }
+                  },
+                  amplitudeEvent: AmplitudeEvents.editPaymentDetailsCanceled,
+                ),
+              );
+            },
+          );
+        },
+      ),
+    );
+  }
+
+  static void show(
+      BuildContext context, List<Member> cachedMembers, VoidCallback onThen) {
```

Comment:
can we have a more describing name than "onThen" pwetty pwease 🥹 something like onBottomSheet... (close, open etc)